astropy/units/tests/test_physical.py

```diff
@@ -1,60 +1,476 @@
+    (length, "length", "__mul__", area),
+    (length, "length", "__rmul__", area),
+    (area, "length", "__truediv__", length),
+    (length, "area", "__rtruediv__", length),
+]
 
 
-def test_flam():
-    flam = u.erg / (u.cm**2 * u.s * u.AA)
-    assert flam.physical_type == 'spectral flux density wav'
+@pytest.mark.parametrize("left, right, operator, expected", operation_parameters)
+def test_physical_type_operations(left, right, operator, expected):
+    """
+    Test that `PhysicalType` dunder methods that require another
+    argument behave as intended.
+    """
+    assert getattr(left, operator)(right) == expected
 
 
-def test_photlam():
-    photlam = u.photon / (u.cm ** 2 * u.s * u.AA)
-    assert photlam.physical_type == 'photon flux density wav'
+unit_with_physical_type_set = [
+    (u.m, {"length"}),
+    (u.kg * u.m / u.s, {"impulse", "momentum"}),
+    (u.Pa, {"energy density", "pressure", "stress"}),
+]
 
 
-def test_photnu():
-    photnu = u.photon / (u.cm ** 2 * u.s * u.Hz)
-    assert photnu.physical_type == 'photon flux density'
+@pytest.mark.parametrize("unit, expected_set", unit_with_physical_type_set)
+def test_physical_type_as_set(unit, expected_set):
+    """Test making a `physical.PhysicalType` instance into a `set`."""
+    resulting_set = set(unit.physical_type)
+    assert resulting_set == expected_set
 
 
-@raises(ValueError)
-def test_redundant_physical_type():
-    physical.def_physical_type(u.m, 'utter craziness')
+def test_physical_type_iteration():
+    """Test iterating through different physical type names."""
+    physical_type_names = [physical_type_name for physical_type_name in pressure]
+    assert physical_type_names == ["energy density", "pressure", "stress"]
 
 
-def test_data_quantity():
-    assert u.byte.physical_type == 'data quantity'
-    assert u.bit.physical_type == 'data quantity'
+def test_physical_type_in():
+    """
+    Test that `in` works as expected for `PhysicalType` objects with one
+    or multiple names.
+    """
+    assert "length" in length
+    assert "pressure" in pressure
+
+
+def test_physical_type_id():
+    """
+    Test that the physical type ID of a `physical.PhysicalType`
+    instance matches that of a unit of that physical type.
+    """
+    time_physical_type = u.s.physical_type
+    time_id = time_physical_type._physical_type_id
+    hour_id = u.hr._get_physical_type_id()
+    assert time_id == hour_id
+
+
+equivalent_unit_pairs = [
+    (u.m, u.m),
+    (u.m, u.cm),
+    (u.N, u.kg * u.m * u.s ** -2),
+    (u.barn * u.Mpc, u.cm ** 3),
+    (u.K, u.deg_C),
+    (u.K, u.imperial.deg_R),
+    (u.K, u.imperial.deg_F),
+    (u.deg_C, u.imperial.deg_F),
+    (u.m ** 18, u.pc ** 18),
+]
+
+
+@pytest.mark.parametrize("unit1, unit2", equivalent_unit_pairs)
+def test_physical_type_id(unit1, unit2):
+    """Test that the physical type IDs of equivalent units match."""
+    id1 = physical.get_physical_type(unit1)._physical_type_id
```

Comment:
Do we really want to test the private `_physical_type_id` attribute rather than just the `PhysicalType` itself? I'm not quite sure what this adds to the tests two functions down.